addons/token-exchange/green_secret_controller.go

```diff
@@ -126,9 +171,23 @@ func validateSecret(secret corev1.Secret) error {
 		return fmt.Errorf("missing storageCluster namespace info in secret %q in namespace %q", secret.Name, secret.Namespace)
 	}
 
+	if string(secret.Data[common.StorageClusterNameKey]) == "" {
```

Comment:
We already have a validate functions in `common` package,`ValidateSourceSecret` and `ValidateDestinationSecret`.

Reply:
Let's clean this in a separate PR.
We need to remove all validations in that case. Not just this one.